### PR TITLE
feat: simplify selectors with data attributes

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

People are already asked to give us unique attributes to look for in the HTML of a page. But, with the new CSS selector library in the toolbar we are seeking for longer selectors to make the unique selectors less fragile. Which means even if `[data-attr="blah"]` is unique we are showing `.SideBar [data-attr="blah"]` as the selector.

## Changes

Since the selector library is already parsing and sanitising the selectors and applying an allow list to which attributes make it into the selector we can be pretty naive in how we check for unique attributes.

So, that's what this does. The simplest possible check to try to present data attributes as unique stand-alone selectors

Screenshot 2023-01-27 at 14 26 48

## How did you test this code?

ran it locally, then duplicated an element with a data-attr and saw that we used the new unique selector and not the data attribute on its own.

### CODE REVIEW

**Comment by mariusandra** [comment]
Just add something like this pseudocode before the call to finder and that's all the complexity we need:

```ts
  for ([attribute, value] in Object.entries(element.dataAttributes)) {
    const selector = `[{escape(attribute)}="{escape(value)}"]`
    if (attribute in allowListedDataAttributes && querySelectorAll(selector).length == 1) {
      return selector
    }
  }

```

File: frontend/src/toolbar/utils.ts [hunk 1]
```diff
@@ -31,11 +31,35 @@ export function elementToQuery(element: HTMLElement, dataAttributes: string[]):
...
+    // check if the data attribute is unique by itself
+    // because the selector string has been through the "finder" library
+    // we can be confident of the format and content
+    const parts = chosenSelector.split(' ')
+    if (parts.length <= 1) {
+        return chosenSelector
+    }
+    const tail = parts[parts.length - 1]
+    return isAttribute(tail) && isUnique(tail) ? tail : chosenSelector
+}
```
[the remark this answers is not in the document]
Comment (pauldambra) [reply frontend/src/toolbar/utils.ts:53]: strings are now forbidden :side-eye-monkey:

File: frontend/src/toolbar/utils.ts [hunk 2]
```diff
@@ -31,35 +31,22 @@ export function elementToQuery(element: HTMLElement, dataAttributes: string[]):
         return
     }
 
-    const chosenSelector = finder(element, {
+    for (const { name, value } of Array.from(element.attributes)) {
+        if (dataAttributes.indexOf(name) === -1) {
```
Comment (mariusandra) [inline frontend/src/toolbar/utils.ts:35]: nit & trick: `haystack.includes(needle)` == `haystack.indexOf(needle) !== -1`

**Review by mariusandra — APPROVED** [review]
Assuming all tests pass 😆  👍

**Pull request merged**